e2e/tests/api-driven/src/demo-workspace/steps/background_steps.ts

```diff
@@ -0,0 +1,134 @@
+import { After, Before, DataTable, Given, World } from "@cucumber/cucumber";
+import {
+  checkTeamsExist,
+  cleanup,
+  createDemoUser,
+  createFlowFromArray,
+  createTeamFromArray,
+  createUser,
+  DataTableArray,
+  DataTableRecord,
+  Flow,
+} from "../helper";
+import { $admin, getClient } from "../../client";
+import { strict as assert } from "node:assert";
+import { CoreDomainClient } from "@opensystemslab/planx-core";
+import { Team, User } from "@opensystemslab/planx-core/types";
+
+export class CustomWorld extends World {
+  demoClient!: CoreDomainClient["client"];
+  adminClient!: CoreDomainClient["client"];
+  demoUser!: Omit<User, "teams">;
+  otherTeam!: DataTableRecord;
+  insertFlowTeamId!: number;
+  demoFlowSlug!: string;
+  adminFlowSlug!: string;
+  demoTeamsArray!: Team[];
+  currentTeamId!: number;
+  teamFlows!: Flow[];
+}
+
+Before("@demo-user-permissions", async function () {
```

Comment:
Ah actually I see you using the tag for cleanup, so my suggestion of removing the duplicates would interfere with this! Feel free to ignore